docs/source/basics/defining_environment.md

```diff
@@ -0,0 +1,239 @@
+
+```R
+install.packages("RCurl")
+install.packages("BiocManager")
+BiocManager::install("S4Vectors")
+```
+</details>
+<br />
+
+<details>
+<summary>Docker commands</summary>
+
+```Dockerfile
+run apt-get update --yes && \
+    apt-get install --yes software-properties-common && \
+    add-apt-repository "deb http://cloud.r-project.org/bin/linux/debian buster-cran40/" && \
+    apt-get install --yes r-base r-base-dev libxml2-dev libcurl4-openssl-dev libssl-dev wget
+copy environment.R /opt/latch/environment.R
+run Rscript /opt/latch/environment.R
+```
+</details>
+<br />
+
+### System (Debian) Packages
+Requirements found in `system-requirements.txt` are installed system-wide using apt.
+
+<details>
+<summary>Example File</summary>
+
+```
+autoconf
+samtools
+```
+</details>
+<br />
+
+
+<details>
+<summary>Docker commands</summary>
+
+```Dockerfile
+copy system-requirements.txt /opt/latch/system-requirements.txt
+run apt-get update --yes && xargs apt-get install --yes </opt/latch/system-requirements.txt
+```
+</details>
+<br />
+
+### Environment Variables
+Environment variables found in an `environment` file will be used to set environment variables inside the workflow environment.
+
+<details>
+<summary>Example File</summary>
+
+```
+BOWTIE2_INDEXES=reference
+PATH="$PATH:/root/bowtie2"
+```
+</details>
+<br />
+
+<details>
+<summary>Docker commands</summary>
+
+```Dockerfile
+env {line1}
+env {line2}
+...
+```
+</details>
+<br />
+
+---
+<br>
+
+To save the generated Dockerfile in the workflow directory, run `latch dockerfile {path_to_workflow_directory}`. The generated Dockefile will be used during subsequent `latch register` and `latch develop` commands to build the workflow environment, and it can be user modified. To go back to the autogenerated Dockerfile, delete the Dockerfile in the workflow directory.
+
+To understand the generated Dockefile, here is an example with instructive comments generated by running `latch init --template subprocess --dockerfile example_workflow`:
+
+```Dockerfile
+# latch base image + dependencies for latch SDK --- removing these will break the workflow
+from 812206152185.dkr.ecr.us-west-2.amazonaws.com/latch-base:ace9-main
+run pip install latch==2.12.1
+run mkdir /opt/latch
+
+# install system requirements
+copy system-requirements.txt /opt/latch/system-requirements.txt
+run apt-get update --yes && xargs apt-get install --yes </opt/latch/system-requirements.txt
+
+# copy all code from package (use .dockerignore to skip files)
+copy . /root/
+
+# set environment variables
+env BOWTIE2_INDEXES=reference
+
+# latch internal tagging system + expected root directory --- changing these lines will break the workflow
+arg tag
+env FLYTE_INTERNAL_IMAGE $tag
+workdir /root
+```
+
+Latch has three base images, one without hardware acceleration drivers, one with CUDA drivers, and one with OPENCL drivers. To use the CUDA or OPENCL base image, use the `--cuda` or `--opencl` flags when running `latch init`.
```

Comment:
- I don't think this should be here, probably just add this info under `latch init`
- this is just an example after all
- will mention this in introduction instead